src/jsutils/uniqueBy.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict
+ */
+
+/**
+ * Returns an array of unique values based on iteratee
+ * which is invoked for each element in array to generate
+ * the criterion by which uniqueness is computed.
+ *
+ * Simiar to _.uniqBy from lodash.
+ */
+export function uniqueBy(
+  array: $ReadOnlyArray<any>,
+  iteratee: (item: any) => any,
+) {
+  const FilteredMap = new Map();
```

Comment:
Why not `Set`?

Reply:
`Set` makes a lot more sense

Reply:
`Map`, `Set` should be used only for non-string keys, `Object.create(null)` is perfectly fine for working with string keys.

Reply:
@IvanGoncharov `Set` is much better *primitive* for checking presence/uniqueness than `Object.create(null)`, both by math sense and by the fact — `Map`s and `Set`s are well tested and perfectly tuned for purpose. Misusing objects as hash maps, well, it's only relique. Polyfilling is not needed today.